src/middlewared/middlewared/plugins/usage.py

```diff
@@ -0,0 +1,416 @@
+from middlewared.schema import accepts
+from middlewared.service import Service, private
+from datetime import datetime
+
+import json
+import asyncio
+import random
+import requests
+
+
+class UsageService(Service):
+    @private
+    async def start(self):
+        try:
+            gather = await self.gather()
```

Comment:
Probably,. if `system.general` -> `crash_reporting` is false, we need to skip gathering and sending data. @william-gr what do you think?

Reply:
Yeah, we need a way to opt-out for this but I am not sure reusing crash_reporting is our best option. Will need to think about it.

Reply:
So we are going to ship 11.3 with this always enabled?

Reply:
No, we are still long ways to 11.3. We will figure something out until then.